File: HrApp_WebAPI.Data/Entities/Employee.cs
using System;
using System.Collections.Generic;

namespace HrApp_WebAPI.Entities
{
    public class Employee
    {
        public int EmployeeId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string PersonalIdentificationNumber { get; set; }
        public DateTime BirthDate { get; set; }
        public string BirthPlace { get; set; }
        //public EmployeePersonalDatas PersonalDatas { get; set; }
        public List<EmployeeAddresses> Addresses { get; set; }
        public List<EmployeeContacts> Contacts { get; set; }
        public List<EmployeeIdentityDocuments> IdentityDocuments { get; set; }
        public List<EmployeeBankData> BankData { get; set; }
        public EmployeeDependents Dependents { get; set; }
        public List<EmployeeContracts> Contracts { get; set; }
        //public Company Company { get; set; }
    }
}

File: HrApp_FrontEnd/src/app/models/employee.ts
import { Company } from "./company";
import { EmployeeAddresses } from "./employeeAddresses";
import { EmployeeBankData } from "./employeeBankData";
import { EmployeeContacts } from "./employeeContacts";
import { EmployeeContracts } from "./employeeContracts";
import { EmployeeDependents } from "./employeeDependents";
import { EmployeeIdentityDocuments } from "./employeeIdentityDocuments";
import { EmployeePersonalDatas } from "./employeePersonalDatas";

export interface Employee {
  employeeId: number;
  firstName: string;
  lastName: string;
  PersonalIdentificationNumber: string;
  BirthDate: Date;
  BirthPlace: string;
  photo: string;
  employeePersonalDatas: EmployeePersonalDatas;
  employeeAddresses: EmployeeAddresses[];
  employeeContacts: EmployeeContacts[];
  employeeIdentityDocuments: EmployeeIdentityDocuments[];
  employeeBankData: EmployeeBankData[];
  employeeDependents: EmployeeDependents;
  employeeContracts: EmployeeContracts[];
  employeeCompany: Company;
}
